derived_types/main.cpp

```diff
@@ -0,0 +1,92 @@
+#include <iostream>
+
+void printArray(int arr[], const int size) {
```

Comment:
I suggest to use construction **type const name** named as "East const" because it's more closer to guidelines and more readable. https://isocpp.org/wiki/faq/const-correctness